### PR TITLE
Various simplifier improvements from dsharletg/sliding-window

### PR DESCRIPTION
**Pull request by dsharletg**
Forking some independent changes from #5755

### CODE REVIEW

File: src/Simplify_Div.cpp [hunk 1]
```diff
@@ -177,9 +180,55 @@ Expr Simplify::visit(const Div *op, ExprInfo *bounds) {
                rewrite((w + (z + (x * c0 + y))) / c1, (y + z + w) / c1 + x * fold(c0 / c1), c0 % c1 == 0 && c1 > 0) ||
                rewrite((w + (z + (y + x * c0))) / c1, (y + z + w) / c1 + x * fold(c0 / c1), c0 % c1 == 0 && c1 > 0) ||
 
-               // Finally, pull out constant additions that are a multiple of the denominator
-               rewrite((x + c0) / c1, x / c1 + fold(c0 / c1), c0 % c1 == 0 && c1 > 0) ||
-               rewrite((c0 - y)/c1, fold(c0 / c1) - y / c1, (c0 + 1) % c1 == 0 && c1 > 0) ||
+               /** In (x + c0) / c1, when can be pull the constant
```
Comment (steven-johnson) [inline src/Simplify_Div.cpp:183]: nit: s/be/we/

File: test/correctness/simplify.cpp [hunk 2]
```diff
@@ -305,6 +305,66 @@ void check_algebra() {
     check((7 - y) / 7, (-y) / 7 + 1);
     check((y - 7) / 7, y / 7 + (-1));
 
+    // TODO: The commented cases below should be handled by
```
Comment (rootjalex) [inline test/correctness/simplify.cpp:308]: Doesn't seem like there are commented cases anymore, so this TODO might be able to be removed?

**Comment by dsharletg** [comment]
Review ping on this... note it's the same as the changes from #5755 @rootjalex @abadams . The buildbot failure is not relevant and I think there is plenty of coverage of this change.

**Review by steven-johnson — APPROVED** [review]

**Review by abadams — APPROVED** [review]
All new rules pass formal verification.

**Pull request merged**